modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -244,6 +244,43 @@ public void shouldCheckAvailability() {
         checkAvailabilityOfDocuments(Arrays.asList(documents.get(1), documents.get(2)), downloadableUntil, 0);
     }
 
+    @Test
+    public void shouldFailUpdateMainContentAlreadyInColdStorage() throws IOException {
+        // move the main content into the cold storage
+        DocumentModel documentModel = createFileDocument(DEFAULT_DOC_NAME, true);
+        moveAndVerifyContent(session, documentModel);
+
+        // we cannot update the main content as it is already in cold storage
+        documentModel = session.getDocument(documentModel.getRef());
+        documentModel.setPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY,
+                (Serializable) Blobs.createBlob(FILE_CONTENT));
+        try {
+            session.saveDocument(documentModel);
+            fail("Should fail because a main content document in cold storage cannot be updated.");
+        } catch (NuxeoException e) {
+            assertEquals(SC_FORBIDDEN, e.getStatusCode());
+            assertEquals(
+                    String.format("The main content of document: %s cannot be updated. It's already in cold storage.",
+                            documentModel.getId()),
+                    e.getMessage());
+        }
+
+        // but we should be able to update the others properties even the attachments
+        documentModel = session.getDocument(documentModel.getRef());
+        documentModel.setPropertyValue("dc:title", "I update the title");
+        documentModel.setPropertyValue("dc:description", "I add a description");
+        var attachments = List.of(Map.of("file", Blobs.createBlob("bar", "text/plain")));
+        documentModel.setPropertyValue("files:files", (Serializable) attachments);
+
+        documentModel = session.saveDocument(documentModel);
+        assertEquals("I update the title", documentModel.getPropertyValue("dc:title"));
+        assertEquals("I add a description", documentModel.getPropertyValue("dc:description"));
+        attachments = (List<Map<String, Blob>>) documentModel.getPropertyValue("files:files");
+        assertEquals(1, attachments.size());
```

Comment:
Could we use `CollecitonUtils.isEqualCollection` here?

Reply:
yep 👍 and i tried using directly the `org.junit.Assert#assertEquals(java.lang.Object, java.lang.Object)` and it works fine. if you agree i can use it.

Reply:
I would suggest using `CollectionUtils` otherwise we're relying on the `equals` method of the asserted object. This could not work as expected, in our case, we probably have an `ArrayList` which has a correct equals implementation. But we shouldn't leverage it.

Reply:
For collections I don't see a problem with using `assertEquals`, their `equals` method are supposed to do the right thing by definition.

Reply:
I agree with you that almost all implementations have a good implementation for `equals`, but not all. Which forces us to use `CollectionUtils` if we want a coding convention.

Reply:
I'm happy to discuss this.